root/opt/puppetlabs/puppet/lib/ruby/2.1.0/facter/mattermost.rb

```diff
@@ -0,0 +1,19 @@
+# mattermost.rb
+
+require 'rubygems'
+
+Facter.add('mattermost') do
+    setcode do
+        mattermost = {}
+        pass = File.read("/var/lib/nethserver/secrets/mattermost").strip
```

Comment:
if mattermost is not installed? we have to check the file before?

Reply:
If mattermost is not installed, the script doesn't exist since it belong to mattermost package.
The password file is generated as soon as mattermost is installed.